pandas/core/resample.py

```diff
@@ -512,11 +512,20 @@ def _wrap_result(self, result):
         """
         Potentially wrap any results.
         """
+        obj = self.obj
+        if (
+            isinstance(result, ABCDataFrame)
+            and result.empty
+            and type(result.index) != PeriodIndex
```

Comment:
This condition seems too specific. Generally specific checks like this mean something down the stack has the bug

Reply:
@mroeschke I have checked it. Passing a df or series with index as PeriodIndex causes problems, and it's handled differently. A similar condition already exists in the same file as the following:

```
    def _wrap_result(self, result):
        result = super()._wrap_result(result)

        # we may have a different kind that we were asked originally
        # convert if needed
        if self.kind == "period" and not isinstance(result.index, PeriodIndex):
            result.index = result.index.to_period(self.freq)
        return result

```

I think it's ok as it gives the required behavior from handling resampling with period index. What do you think?

Reply:
@mroeschke do you think it needs another PR? I feel it will be a more influential edit that causes both #47705 and #47787

Reply:
Feel free to just use this PR to make the change.

Reply:
@mroeschke Done